### ISSUE TITLE
`string_lit_as_bytes` gives incorrect recommendation with stringify!()

### ISSUE THREAD

**Issue by ipetkov** [Manishearth/rust-clippy#763]
Sample program:

```rust
#![feature(plugin)]
#![plugin(clippy)]

fn main() {
    let _ = bstringify!(foo).as_bytes();
}
```

Warning given:
> calling `as_bytes()` on a string literal. Consider using a byte string literal instead: `bstringify!(foo)`, #[warn(string_lit_as_bytes)] on by default

Of course, using `bstringify!(foo)` gives an undefined macro error from rustc.

**Comment by ipetkov** [comment]
Not sure if the lint can even recognize this situation as the macro is probably expanded to a string literal by then, but it did get me to momentarily believe there was a bstringify! macro I was not aware of.

**Comment by llogiq** [comment]
We can add a macro check to the `ExprLit`.
